addons/custom-menu-bar/username.js

```diff
@@ -1,14 +1,13 @@
 export default async function ({ addon }) {
-  if (await addon.auth.fetchIsLoggedIn()) {
-    const username = await addon.auth.fetchUsername();
-    addon.tab.redux.initialize(); // Start listening to Redux events
-    addon.tab.redux.addEventListener("statechanged", async (e) => {
-      if (!addon.settings.get("compact-username")) return; // Skip if the option is disabled
-      if (e.detail.action.type !== "scratch-gui/menus/OPEN_MENU" || e.detail.action.menu !== "accountMenu") return; // Skip if another Redux event occurs
-      const profileSpans = await addon.tab.waitForElement(".menu_menu-item_3EwYA.menu_hoverable_3u9dt"); // Do this if the menu is opened
-      const profile = profileSpans.appendChild(document.createElement("span")); // Add username to "Profile" menu option
-      profile.textContent = username;
-      profile.className = "sa-profile-name";
+  while (true) {
+    const menuItem = await addon.tab.waitForElement("[class^=menu_menu_] > :first-child", {
```

Comment:
```suggestion
    const menuItem = await addon.tab.waitForElement("[class^=menu-bar_account-info-group_] [class^=menu_menu_] > :first-child", {
```
`reduxCondition` is only meant to be used as an optimization - the selector should still work without the condition. Currently, this happens if the condition is removed:
![image](https://github.com/ScratchAddons/ScratchAddons/assets/51849865/3561db74-6133-4674-868b-ee50ecc8dc69)

Reply:
Indeed, it shouldn't match more elements than needed if reduxCondition theoretically always evaluates to true.